src/virtual_dom/vlist.rs

```diff
@@ -104,22 +105,67 @@ impl VDiff for VList {
 
         // Process children
         let mut lefts = self.children.iter_mut();
-        let mut rights = rights.drain(..);
-        loop {
-            match (lefts.next(), rights.next()) {
-                (Some(left), Some(right)) => {
-                    previous_sibling = left.apply(parent, previous_sibling.as_ref(), Some(right));
+        let has_keys = if let Some(first) = rights.first() {
+            if let Some(_) = first.key() {
+                true
+            } else {
+                false
+            }
+        } else {
+            true
+        };
+        if has_keys {
+            let mut rights_lookup = HashMap::with_capacity(rights.len());
+            for r in rights.drain(..) {
+                if r.key().is_none() {
+                    log::warn!(
```

Comment:
It's not necessarily a bug, you could have: 
```rust
html! {
  <>
    <h1>Hello</h1>
    <div key={"1"} />
    <div key={"2'} />
  </>
}

Reply:
Fixed by switching reverting to a vector if the node doesn't have a key.